sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -94,6 +94,12 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     case Distinct(p: Project) =>
       projectToSQL(p, isDistinct = true)
 
+    case p@ Project(_, g: Generate) =>
+      generateToSQL(p)
+
+    case g: Generate =>
```

Comment:
not quite sure, but can we convert this to `Project(g.outpur, g)` and share the same code path with the above case?